lib/std/Build/Watch.zig

```diff
@@ -237,6 +237,300 @@ const Os = switch (builtin.os.tag) {
             }
         }
     },
+    .windows => struct {
+        const posix = std.posix;
+        const windows = std.os.windows;
+
+        /// Keyed differently but indexes correspond 1:1 with `dir_table`.
+        handle_table: HandleTable,
+        dir_list: std.ArrayListUnmanaged(*Directory),
+        io_cp: ?windows.HANDLE,
+
+        const HandleTable = std.ArrayHashMapUnmanaged(FileId, ReactionSet, FileId.Adapter, false);
+
+        const FileId = struct {
+            volumeSerialNumber: windows.ULONG,
+            indexNumber: windows.LARGE_INTEGER,
+
+            const Adapter = struct {
+                pub fn hash(self: Adapter, a: FileId) u32 {
+                    _ = self;
+                    var hasher = Hash.init(0);
+                    std.hash.autoHash(&hasher, a);
+                    return @truncate(hasher.final());
+                }
+                pub fn eql(self: Adapter, a: FileId, b: FileId, b_index: usize) bool {
+                    _ = self;
+                    _ = b_index;
+                    return a.volumeSerialNumber == b.volumeSerialNumber and a.indexNumber == b.indexNumber;
+                }
+            };
+        };
```

Comment:
I can just use `AutoArrayHashMapUnmanaged` and delete this `FileId.Adapter` stuff?